ArtOfIllusion/src/artofillusion/procedural/IOPort.java

```diff
@@ -134,14 +137,35 @@ public void draw(Graphics g)
       g.setColor(Color.BLACK);
     else
       g.setColor(Color.BLUE);
-    if (location == TOP)
-      g.fillPolygon(new int[] {x+SIZE, x-SIZE, x}, new int[] {y, y, y+SIZE}, 3);
-    else if (location == BOTTOM)
-      g.fillPolygon(new int[] {x+SIZE, x-SIZE, x}, new int[] {y, y, y-SIZE}, 3);
-    else if (location == LEFT)
-      g.fillPolygon(new int[] {x, x, x+SIZE}, new int[] {y+SIZE, y-SIZE, y}, 3);
-    else if (location == RIGHT)
-      g.fillPolygon(new int[] {x-SIZE, x-SIZE, x}, new int[] {y+SIZE, y-SIZE, y}, 3);
+    
+    switch (location) {
+      case TOP:
+        g.fillPolygon(new int[] {x+SIZE, x-SIZE, x}, new int[] {y, y, y+SIZE}, 3);
+        break;
+      case BOTTOM:
+        g.fillPolygon(new int[] {x+SIZE, x-SIZE, x}, new int[] {y, y, y-SIZE}, 3);
+        break;
+      case LEFT:
+        g.fillPolygon(new int[] {x, x, x+SIZE}, new int[] {y+SIZE, y-SIZE, y}, 3);
+        break;
+     case RIGHT:
+        g.fillPolygon(new int[] {x-SIZE, x-SIZE, x}, new int[] {y+SIZE, y-SIZE, y}, 3);
+        break;
+     default:
+        break;
+      }
   }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("IOPort{valueType=").append(valueType == COLOR ? "Color" : "Number");
+        sb.append(", type=").append(type == INPUT ? "Input" : "Output");
+        sb.append(", description=").append(Arrays.toString(description));
+        sb.append('}');
+        return sb.toString();
+    }
+  
+  
```

Comment:
Does this have any use outside of debugging? If not, it belongs in the TestClass that uses it.